pydicom/pixel_data_handlers/rle_handler.py

```diff
@@ -203,17 +203,18 @@ def _rle_decode_plane(data):
 
     while pos < len_data:
         header_byte = data[pos]
-
         pos += 1
-
         if header_byte > 128:
-            result.extend([data[pos]] * (255 - header_byte + 2))
+            # Extend by copying the next byte (-N + 1) times
+            # however since using uint8 instead of int8 this will be
+            # (256 - N + 1) times
+            result.extend(data[pos:pos + 1] * (257 - header_byte))
             pos += 1
```

Comment:
I don't see this making any difference on 2.7, but I think the change is clearer and at least no slower.